poweriq-worker/src/main/java/com/vmware/flowgate/poweriqworker/jobs/PowerIQService.java

```diff
@@ -564,16 +561,111 @@ public void saveSensorAssetsToFlowgate(Map<String, Asset> exsitingSensorAssets,
                   asset.setMountingSide(sensorMountingSide.get(sensor.getPosition().toUpperCase()));
                }
                //save
-               assetsNeedToSave.add(asset);
+               newAssetsNeedToSave.add(asset);
             }
          }
-         restClient.saveAssets(assetsNeedToSave);
+         restClient.saveAssets(oldAssetsNeedToupdate);
+         if(!newAssetsNeedToSave.isEmpty()) {
+            List<Asset> sensorAlreadySaved = Arrays.asList(restClient.saveAssets(newAssetsNeedToSave).getBody());
+            List<Asset> pduAssetNeedToUpdate = updatePduMetricformular(sensorAlreadySaved,pduAssetMap);
+            restClient.saveAssets(pduAssetNeedToUpdate);
+         }
          offset += limit;
       }
+   }
+
+   public List<Asset> updatePduMetricformular(List<Asset> sensorAssets, Map<String,Asset> pduIdAndAssetMap){
+      List<Asset> pduAssets = new ArrayList<Asset>();
+      for(Asset sensorAsset : sensorAssets) {
+         com.vmware.flowgate.common.model.Parent parent = sensorAsset.getParent();
+         if(parent == null) {
+            continue;
+         }
+         String pduId = parent.getParentId();
+         Asset pduAsset = pduIdAndAssetMap.get(pduId);
+         int position = sensorAsset.getCabinetUnitPosition();
```

Comment:
I don't think you can use the cabinetUnitionPosition of sensor here , since it will always 0. 

Take PowerIQ for example, it's position only have inlet, outlet ,external. do you have  code that  save those position information?

Reply:
Updated